packages/moonstone/VideoPlayer/VideoPlayer.js

```diff
@@ -1952,17 +1964,29 @@ const VideoPlayerBase = class extends React.Component {
 				style={style}
 			>
 				{/* Video Section */}
-				<Media
-					{...rest}
-					autoPlay={!noAutoPlay}
-					className={css.video}
-					component={videoComponent}
-					controls={false}
-					onUpdate={this.handleEvent}
-					ref={this.setVideoRef}
-				>
-					{source}
-				</Media>
+				{this.props.preloadSource ?
```

Comment:
I'm curious why this check happens here, instead of inside the `VideoWithPreload` component, and more fundamentally, why `VideoWithPreload` is different from `Media`. Shouldn't `Media` just have a `preloadSource` prop, which then invokes that line of logic inside `Media`? One component able to support preloading or normal.